Loop/Managers/LoopDataManager.swift

```diff
@@ -1002,17 +1006,17 @@ extension LoopDataManager {
         var retrospectiveGlucoseEffect = self.retrospectiveGlucoseEffect
         var effects: [[GlucoseEffect]] = []
 
-        if inputs.contains(.carbs), let carbEffect = self.carbEffect {
-            if let potentialCarbEntry = potentialCarbEntry, var recentEntries = recentCarbEntries {
-                if let replacedCarbEntry = replacedCarbEntry, let index = recentEntries.firstIndex(of: replacedCarbEntry) {
-                    recentEntries.remove(at: index)
-                }
-
+        if inputs.contains(.carbs) {
```

Comment:
Ah, I forgot to leave a comment here; thanks for calling this out. 

I had managed to contrive a scenario where, in the absence of historical carb data (e.g. a fresh install), `recentCarbEntries` was `nil` even though `potentialCarbEntry` was non-`nil`, which (from the branch that occurred on 1006 prior to this commit) resulted in the carb effect of the potential carb entry not being included in the prediction.

This group of changes ensures that, regardless of whether `carbEffect` or `recentCarbEntries` is `nil`, the carb effect of `potentialCarbEntry` is always computed and included. The bottom lines make an analogous change for `insulinEffect` and `potentialBolus`.

Reply:
This circumstance only manifested during manual glucose entry, since in other contexts this function would only be run while loop data was valid, which meant that `recentCarbEntries` was always assigned (even if just an empty array).

Reply:
I think this will need approval from @ps2. I don't have enough understanding at this point.